### PR TITLE
Add `RGBLIGHT_DEFAULT_ON` macro configuration option

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds the possibility of defining RGB lighting disabled by deafult at  EEPROM initialization.

This is done through a macro in `config.h`:

```c
#define RGBLIGHT_DEFAULT_ON false
```

The macro works in the `rgblight.c` quantum driver setting enable to true if the macro is not enabled and false if it is.

An accompanying addition to the documentation is also pushed.

This was tested on a couple PCBs around; all STM32 though, but I am sure this should not pose an  issue for the AVR units.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Review by fauxpark — COMMENTED** [review]
I'm thinking this should be inverted and default to `true` to be consistent with the other defines:
```c
#if !defined(RGBLIGHT_DEFAULT_ON)
#    define RGBLIGHT_DEFAULT_ON true
#endif
```

**Comment by Gondolindrim** [comment]
> I'm thinking this should be inverted and default to `true` to be consistent with the other defines:
> 
> ```c
> #if !defined(RGBLIGHT_DEFAULT_ON)
> #    define RGBLIGHT_DEFAULT_ON true
> #endif
> ```

So the idea would be, if I want the RGB off by default I'd use

```c
#undef RGBLIGHT_DEFAULT_ON
```

?

**Comment by fauxpark** [comment]
No, you'd set it to `false`.

**Comment by Gondolindrim** [comment]
> No, you'd set it to `false`.

Done in https://github.com/qmk/qmk_firmware/pull/20857/commits/c253989fe83ede7e3e7c11e22a55b6b28e5dbb17

File: docs/feature_rgblight.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ Your RGB lighting can be configured by placing these `#define`s in your `config.
 |`RGBLIGHT_DEFAULT_SAT`     |`UINT8_MAX` (255)           |The default saturation to use upon clearing the EEPROM                                                                     |
 |`RGBLIGHT_DEFAULT_VAL`     |`RGBLIGHT_LIMIT_VAL`        |The default value (brightness) to use upon clearing the EEPROM                                                             |
 |`RGBLIGHT_DEFAULT_SPD`     |`0`                         |The default speed to use upon clearing the EEPROM                                                                          |
+|`RGBLIGHT_DEFAULT_ON`      |`1`                         |If defined  `0`, RGB lighting will be disabled (off) by default when EEPROM is initialized.                                |
```
Comment (fauxpark) [inline docs/feature_rgblight.md:105]:
```suggestion
|`RGBLIGHT_DEFAULT_ON`      |`true`                      |Enable RGB lighting upon clearing the EEPROM                           |
```

File: quantum/rgblight/rgblight.c [hunk 2]
```diff
@@ -89,6 +89,11 @@ static uint8_t mode_base_table[] = {
 #    define RGBLIGHT_DEFAULT_SPD 0
 #endif
 
+// If RGBLIGHT_DEFAULT_ON is set 0, RGBLIGHT will be disabled at EEPROM initialization.
+#if !defined(RGBLIGHT_DEFAULT_ON)
+#    define RGBLIGHT_DEFAULT_ON 1
```
Comment (fauxpark) [inline quantum/rgblight/rgblight.c:94]:
```suggestion
#    define RGBLIGHT_DEFAULT_ON true
```

File: quantum/rgblight/rgblight.c [hunk 3]
```diff
@@ -198,12 +203,16 @@ void eeconfig_update_rgblight_current(void) {
 }
 
 void eeconfig_update_rgblight_default(void) {
+#if RGBLIGHT_DEFAULT_ON
     rgblight_config.enable = 1;
```
Comment (fauxpark) [inline quantum/rgblight/rgblight.c:207]:
```suggestion
    rgblight_config.enable = RGBLIGHT_DEFAULT_ON;
```

File: quantum/rgblight/rgblight.c [hunk 4]
```diff
@@ -198,12 +203,16 @@ void eeconfig_update_rgblight_current(void) {
...
-    rgblight_config.val    = RGBLIGHT_DEFAULT_VAL;
-    rgblight_config.speed  = RGBLIGHT_DEFAULT_SPD;
+#else
+    rgblight_config.enable = 0;
+#endif
+    rgblight_config.mode  = RGBLIGHT_DEFAULT_MODE;
+    rgblight_config.hue   = RGBLIGHT_DEFAULT_HUE;
+    rgblight_config.sat   = RGBLIGHT_DEFAULT_SAT;
+    rgblight_config.val   = RGBLIGHT_DEFAULT_VAL;
+    rgblight_config.speed = RGBLIGHT_DEFAULT_SPD;
```
Comment (fauxpark) [inline quantum/rgblight/rgblight.c:215]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: docs/feature_rgblight.md [hunk 5]
```diff
@@ -102,6 +102,7 @@ Your RGB lighting can be configured by placing these `#define`s in your `config.
 |`RGBLIGHT_DEFAULT_SAT`     |`UINT8_MAX` (255)           |The default saturation to use upon clearing the EEPROM                                                                     |
 |`RGBLIGHT_DEFAULT_VAL`     |`RGBLIGHT_LIMIT_VAL`        |The default value (brightness) to use upon clearing the EEPROM                                                             |
 |`RGBLIGHT_DEFAULT_SPD`     |`0`                         |The default speed to use upon clearing the EEPROM                                                                          |
+|`RGBLIGHT_DEFAULT_ON`      |`true`                      |Enable RGB lighting upon clearing the EEPROM                           |
```
Comment (fauxpark) [inline docs/feature_rgblight.md:105]:
```suggestion
|`RGBLIGHT_DEFAULT_ON`      |`true`                      |Enable RGB lighting upon clearing the EEPROM                                                                               |
```

File: quantum/rgblight/rgblight.c [hunk 6]
```diff
@@ -89,6 +89,11 @@ static uint8_t mode_base_table[] = {
 #    define RGBLIGHT_DEFAULT_SPD 0
 #endif
 
+// If RGBLIGHT_DEFAULT_ON is set 0, RGBLIGHT will be disabled at EEPROM initialization.
```
Comment (fauxpark) [inline quantum/rgblight/rgblight.c:92]:
```suggestion
```

File: quantum/rgblight/rgblight.c [hunk 7]
```diff
@@ -198,12 +203,7 @@ void eeconfig_update_rgblight_current(void) {
...
 
 void eeconfig_update_rgblight_default(void) {
-    rgblight_config.enable = 1;
-    rgblight_config.mode   = RGBLIGHT_DEFAULT_MODE;
-    rgblight_config.hue    = RGBLIGHT_DEFAULT_HUE;
-    rgblight_config.sat    = RGBLIGHT_DEFAULT_SAT;
-    rgblight_config.val    = RGBLIGHT_DEFAULT_VAL;
-    rgblight_config.speed  = RGBLIGHT_DEFAULT_SPD;
+    rgblight_config.enable = RGBLIGHT_DEFAULT_ON;
     RGBLIGHT_SPLIT_SET_CHANGE_MODEHSVS;
```
Comment (fauxpark) [inline quantum/rgblight/rgblight.c:207]:
```suggestion
    rgblight_config.mode   = RGBLIGHT_DEFAULT_MODE;
    rgblight_config.hue    = RGBLIGHT_DEFAULT_HUE;
    rgblight_config.sat    = RGBLIGHT_DEFAULT_SAT;
    rgblight_config.val    = RGBLIGHT_DEFAULT_VAL;
    rgblight_config.speed  = RGBLIGHT_DEFAULT_SPD;
    RGBLIGHT_SPLIT_SET_CHANGE_MODEHSVS;
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**